Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/XLGppDataStore.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.ContractsLight;
+using BuildXL.Engine.Cache.KeyValueStores;
+using BuildXL.Execution.Analyzer;
+using BuildXL.Scheduler.Tracing;
+using BuildXL.Utilities;
+using Google.Protobuf;
+
+namespace BuildXL.Analyzers.Core.XLGPlusPlus
+{
+    public class XLGppDataStore
+    {
+        /// <summary>
+        /// Rocks DB Accessor for XLG++ data
+        /// </summary>
+        public KeyValueStoreAccessor Accessor { get; set; }
+
+        public XLGppDataStore()
+        {
+
+        }
+
+        /// <summary>
+        /// Open the datastore and populate the KeyValueStoreAccessor for the XLG++ DB
+        /// </summary>
+        /// <returns>Boolean if datastore was opened successfully</returns>
+        public bool OpenDatastore(string storeDirectory,
+            bool defaultColumnKeyTracked = false,
+            IEnumerable<string> additionalColumns = null,
+            IEnumerable<string> additionalKeyTrackedColumns = null,
+            Action<Failure> failureHandler = null,
+            bool openReadOnly = false,
+            bool dropMismatchingColumns = false,
+            bool onFailureDeleteExistingStoreAndRetry = false)
+        {
+
+            var accessor = KeyValueStoreAccessor.Open(storeDirectory, 
+                defaultColumnKeyTracked, 
+                additionalColumns, 
+                additionalKeyTrackedColumns, 
+                failureHandler, 
+                openReadOnly, 
+                dropMismatchingColumns, 
+                onFailureDeleteExistingStoreAndRetry);
+
+            if (accessor.Succeeded)
+            {
+                Accessor = accessor.Result;
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Gets all the events of a certain type from the DB
+        /// </summary>
+        /// <returns>List of event objects recovered from DB </returns>
+        public List<string> GetEventsByType(int eventTypeID)
+        {
+            Contract.Assert(Accessor != null, "XLGppStore must be initialized via OpenDatastore first");
+
+            var storedEvents = new List<string>();
+            var eventQuery = new EventTypeQuery
+            {
+                EventTypeID = eventTypeID,
+            };
+            Analysis.IgnoreResult(
+                Accessor.Use(database =>
+                {
+                    foreach (var kvp in database.PrefixSearch(eventQuery.ToByteArray()))
+                    {
+                        Console.WriteLine(BXLInvocationEvent.Parser.ParseFrom(kvp.Value));

```

Comment:
> Console.WriteLine(BXLInvocationEvent.Parser.ParseFrom(kvp.Value)); [](start = 23, length = 67)

if you really want an option for this to be printed on the console, you need to hide it behind some flag.

Reply:
The idea for this was to just print to console for now to validate that the function works as desired. 

---
In reply to: [306052610](https://github.com/microsoft/BuildXL/pull/636#discussion_r306052610) [](ancestors = 306052610)

Reply:
I think that's fine for this PR. In the next one this can be superseded with tests that programmatically validate things

---
In reply to: [306066481](https://github.com/microsoft/BuildXL/pull/636#discussion_r306066481) [](ancestors = 306066481,306052610)